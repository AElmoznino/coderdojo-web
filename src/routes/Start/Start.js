// @flow

import React from 'react'
import GridColumn from 'components/Grid/GridColumn'
import StartSection from './components/StartSection'
import { Query } from 'react-apollo'
import gql from 'graphql-tag'

const GET_LEVELS = gql`
  {
    levels {
      link
      title
      text
    }
  }
`

export const Start = () => (
  <Query query={GET_LEVELS}>
    {({ data, error, loading }) => {
      if (loading) return <p>Loading...</p>
      if (error) return <p>Error :(</p>
      if (!data) return null

      return (
        <GridColumn>
          {data.levels.map(level => (
            <StartSection
              key={level.title}
              link={level.link}
              text={level.text}
              title={level.title}
            />
          ))}
        </GridColumn>
      )
    }}
  </Query>
)

export default Start
