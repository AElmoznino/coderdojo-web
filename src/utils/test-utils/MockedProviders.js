import * as React from 'react'
import { ApolloProvider } from 'react-apollo'
import { BrowserRouter } from 'react-router-dom'
import { ThemeProvider } from 'styled-components'
import { theme } from '../../theme'
import ApolloClient from 'apollo-boost'
import createMockedNetworkFetch from 'apollo-mocknetworkinterface'

jest.unmock('react-apollo')
jest.unmock('react-router-dom')

type MockedProvidersProps = {
  children: React.Node,
  loading?: boolean,
  response?: Object,
}

const MockedProviders = ({
  children,
  loading,
  response = {},
}: MockedProvidersProps) => {
  const createResponse = () => {
    return {
      loading: !!loading,
      data: {
        ...response,
      },
    }
  }

  const mockedNetworkFetch = createMockedNetworkFetch(createResponse, {
    latency: 0,
  })

  const client = new ApolloClient({
    uri: 'http://localhost:4000',
    fetch: mockedNetworkFetch,
  })

  return (
    <ApolloProvider client={client}>
      <ThemeProvider theme={theme}>
        <BrowserRouter>{children}</BrowserRouter>
      </ThemeProvider>
    </ApolloProvider>
  )
}

export default MockedProviders
