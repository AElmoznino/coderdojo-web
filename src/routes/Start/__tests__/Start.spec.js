import React from 'react'
import { mount } from 'enzyme'
import Start from '../Start'
import MockedProviders from 'utils/test-utils/MockedProviders'

jest.unmock('graphql-tag')
jest.unmock('react-apollo')

describe('components/Start', () => {
  let component
  let mockedResponse

  beforeEach(() => {
    mockedResponse = {
      levels: [
        {
          __typename: 'Level',
          link: '/overview/nybörjare',
          title: 'Nybörjare',
          text:
            'Har du aldrig kodat förrut? Då börjar du här och får en introduktion\n          till hur man strukturerar upp hemsidor med HTML och snyggar till dem\n          med CSS.',
        },
        {
          __typename: 'Level',
          link: '/overview/fortsättning',
          title: 'Fortsättning',
          text:
            'Har du redan prövat lite skriven programmering (dvs inte code.org eller Scratch)\n    så är det här du fortsätter. Du får pröva på Testdriven utveckling och att ändra dina\n    hemsidor med JavaScript',
        },
      ],
    }

    component = mount(
      <MockedProviders response={mockedResponse}>
        <Start />
      </MockedProviders>,
    )
  })

  it('renders Start', done => {
    setImmediate(() => {
      component.update()

      expect(component).toMatchSnapshot()

      done()
    })
  })
})
