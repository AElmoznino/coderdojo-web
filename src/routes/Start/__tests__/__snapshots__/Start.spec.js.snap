// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Start renders Start 1`] = `
<styled.div>
  <div
    className="sc-bdVaJa cBlAJt"
  >
    <styled.div>
      <div
        className="sc-bxivhb dosbik"
      >
        <styled.h1>
          <h1
            className="sc-htpNat kpiFDp"
          >
            Nybörjare
          </h1>
        </styled.h1>
        <styled.div>
          <div
            className="sc-ifAKCX eEDyMQ"
          >
            <img
              alt="placeholder alt"
              src="https://placeimg.com/200/200/tech"
            />
            <styled.div>
              <div
                className="sc-EHOje loaWLG"
              >
                <p>
                  Har du aldrig kodat förrut? Då börjar du här och får en introduktion
          till hur man strukturerar upp hemsidor med HTML och snyggar till dem
          med CSS.
                </p>
                <Styled(Styled(Link))
                  to="/overview/nybörjare"
                >
                  <a
                    className="sc-bwzfXH sc-bZQynM eJlLuS"
                    href="/overview/nybörjare"
                    onClick={[Function]}
                  >
                    Kom igång
                  </a>
                </Styled(Styled(Link))>
              </div>
            </styled.div>
          </div>
        </styled.div>
      </div>
    </styled.div>
    <styled.div>
      <div
        className="sc-bxivhb dosbik"
      >
        <styled.h1>
          <h1
            className="sc-htpNat kpiFDp"
          >
            Fortsättning
          </h1>
        </styled.h1>
        <styled.div>
          <div
            className="sc-ifAKCX eEDyMQ"
          >
            <img
              alt="placeholder alt"
              src="https://placeimg.com/200/200/tech"
            />
            <styled.div>
              <div
                className="sc-EHOje loaWLG"
              >
                <p>
                  Har du redan prövat lite skriven programmering (dvs inte code.org eller Scratch)
    så är det här du fortsätter. Du får pröva på Testdriven utveckling och att ändra dina
    hemsidor med JavaScript
                </p>
                <Styled(Styled(Link))
                  to="/overview/fortsättning"
                >
                  <a
                    className="sc-bwzfXH sc-bZQynM eJlLuS"
                    href="/overview/fortsättning"
                    onClick={[Function]}
                  >
                    Kom igång
                  </a>
                </Styled(Styled(Link))>
              </div>
            </styled.div>
          </div>
        </styled.div>
      </div>
    </styled.div>
  </div>
</styled.div>
`;
